sdk/identity/azure-identity/src/default_azure_credential.cpp

```diff
@@ -38,17 +43,46 @@ DefaultAzureCredential::DefaultAzureCredential(
             "is the better fit for the application.");
 
   // Creating credentials in order to ensure the order of log messages.
-  auto const envCred = std::make_shared<EnvironmentCredential>(options);
-  auto const wiCred = std::make_shared<WorkloadIdentityCredential>(options);
-  auto const azCliCred = std::make_shared<AzureCliCredential>(options);
-  auto const managedIdentityCred = std::make_shared<ManagedIdentityCredential>(options);
+  ChainedTokenCredential::Sources miSources;
+  {
+    miSources.emplace_back(std::make_shared<EnvironmentCredential>(options));
+    miSources.emplace_back(std::make_shared<WorkloadIdentityCredential>(options));
+
+    constexpr auto envVarName = "AZURE_TOKEN_CREDENTIALS";
+    const auto envVarValue = Environment::GetVariable(envVarName);
+
+    const auto isProd = StringExtensions::LocaleInvariantCaseInsensitiveEqual(envVarValue, "prod");
```

Comment:
As was done in other languages, trim any leading or trailing whitespace from the env var value before the string comparison.

Reply:
Thank you, fixed.